### PR TITLE
[move] Rework of Call and Module Access syntax

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- The grammar for call syntax and identifiers was extremely complicated
- Additionally, methods were a sort of half baked features, and didn't let you add `impl` style methods for custom instantiations, i.e. no methods for `impl Vec.T<Coin>` or anything specific. 
- Module access is now :: instead of .
- Removed MName vs Name restrictions on identifiers
  -  e.g. structs can start with lower, functions can start with upper
- Killed method syntax
- Changed call syntax to require parens
  - Will change some typing rules around this later

## Test Plan

Updated all tests

### CODE REVIEW

File: language/move-lang/stdlib/modules/event.move [hunk 1]
```diff
@@ -25,25 +25,25 @@ module Event {
         guid: bytearray,
     }
 
-    // Derive a fresh unique id by using sender's HandleIdGenerator. The generated bytearray is indeed unique because it
+    // Derive a fresh unique id by using sender's HandleIdGenerator:: The generated bytearray is indeed unique because it
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/stdlib/modules/event.move]: Ooops ...

File: language/move-lang/tests/move_check/naming/invalid_call_lhs_parens_around_name.exp [hunk 2]
```diff
@@ -1,11 +1,11 @@
 error: 
 
-   ┌── tests/move_check/naming/invalid_call_lhs_parens_around_name.move:3:9 ───
+   ┌── tests/move_check/naming/invalid_call_lhs_parens_around_name.move:3:14 ───
    │
  3 │         (foo)()
-   │         ^^^^^^^ Invalid function call
+   │              ^^ Unrecognized Token: ()
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/naming/invalid_call_lhs_parens_around_name.exp]: This is erroring correctly in parsing. I should probably move the test.

File: language/move-lang/tests/move_check/naming/invalid_call_lhs_complex_expression.exp [hunk 3]
```diff
@@ -1,22 +1,11 @@
 error: 
 
-   ┌── tests/move_check/naming/invalid_call_lhs_complex_expression.move:3:9 ───
+   ┌── tests/move_check/naming/invalid_call_lhs_complex_expression.move:3:29 ───
    │
  3 │         (if (true) 5 else 0)();
-   │         ^^^^^^^^^^^^^^^^^^^^^^ Invalid function call
+   │                             ^^ Unrecognized Token: ()
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/naming/invalid_call_lhs_complex_expression.exp]: This is erroring correctly in parsing. I should probably move the test.

File: language/move-lang/tests/move_check/naming/invalid_call_lhs_return.exp [hunk 4]
```diff
@@ -1,11 +1,11 @@
 error: 
 
-   ┌── tests/move_check/naming/invalid_call_lhs_return.move:3:9 ───
+   ┌── tests/move_check/naming/invalid_call_lhs_return.move:3:20 ───
    │
  3 │         (return ())(0, 1);
-   │         ^^^^^^^^^^^^^^^^^ Invalid function call
+   │                    ^ Unrecognized Token: (
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/naming/invalid_call_lhs_return.exp]: This is erroring correctly in parsing. I should probably move the test.

File: language/move-lang/tests/move_check/naming/invalid_call_lhs_value.exp [hunk 5]
```diff
@@ -1,22 +1,11 @@
 error: 
 
-   ┌── tests/move_check/naming/invalid_call_lhs_value.move:3:9 ───
+   ┌── tests/move_check/naming/invalid_call_lhs_value.move:3:10 ───
    │
  3 │         5();
-   │         ^^^ Invalid function call
+   │          ^^ Unrecognized Token: ()
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/naming/invalid_call_lhs_value.exp]: This is erroring correctly in parsing. I should probably move the test.

File: language/move-lang/tests/move_check/expansion/invalid_unpack_assign_lhs_mdot_no_addr.exp [hunk 6]
```diff
@@ -1,24 +1,11 @@
 error: 
 
-   ┌── tests/move_check/expansion/invalid_unpack_assign_lhs_mdot_no_addr.move:4:9 ───
+   ┌── tests/move_check/expansion/invalid_unpack_assign_lhs_mdot_no_addr.move:4:14 ───
    │
- 4 │         false.M { f } = 0;
-   │         ^^^^^^^ Invalid module type name access. Expected: a (possibly qualified) struct name
+ 4 │         false::M { f } = 0;
+   │              ^^ Unrecognized Token: ::
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/expansion/invalid_unpack_assign_lhs_mdot_no_addr.exp]: Yes

File: language/move-lang/tests/move_check/expansion/invalid_unpack_assign_rhs_not_fields.exp [hunk 7]
```diff
@@ -1,24 +1,11 @@
...
-
-error: 
-
-    ┌── tests/move_check/expansion/invalid_unpack_assign_rhs_not_fields.move:13:13 ───
+    ┌── tests/move_check/expansion/invalid_unpack_assign_rhs_not_fields.move:11:14 ───
     │
- 13 │         X.S { 0 } = 0;
-    │             ^^^^^ Invalid fields for deconstructing assignment. Expected: named arguments
+ 11 │         X::S 0 = 0;
+    │              ^ Unrecognized Token: 0
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/expansion/invalid_unpack_assign_rhs_not_fields.exp]: Yes

File: language/move-lang/tests/move_check/expansion/invalid_pack_mname_non_addr.exp [hunk 8]
```diff
@@ -1,16 +1,11 @@
...
-   │
-
-error: 
-
-   ┌── tests/move_check/expansion/invalid_pack_mname_non_addr.move:8:9 ───
-   │
- 8 │         bar().bar().M.S { }
-   │         ^^^^^^^^^^^ Invalid fully qualified module type name. Expected: an address
+ 4 │         false.M::S { }
+   │                ^^ Unrecognized Token: ::
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply language/move-lang/tests/move_check/expansion/invalid_pack_mname_non_addr.exp]: its a weird one.The old version used to be testing that I was converting dot chains to name access properly. So the new version doesn't have the same analogous test

File: language/move-lang/tests/move_check/naming/invalid_call_lhs_complex_expression.exp (hunk 3 above)
[the remark this answers is not in the document]
Comment (bob-wilson) [reply language/move-lang/tests/move_check/naming/invalid_call_lhs_complex_expression.exp]: That's a pretty terrible error message, but that's a separate topic.....

**Pull request closed without merging**